pkgs/applications/video/corrscope/default.nix

```diff
@@ -56,6 +59,12 @@ mkDerivationWith python3Packages.buildPythonApplication rec {
     )
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = corrscope;
+    # Tries writing to $HOME/.local/share/corrscope
+    command = "env XDG_DATA_HOME=$TMPDIR ${lib.getExe corrscope} --version";
```

Comment:
```
building '/nix/store/2vzb1xjr9hk6sjxf32j1ph856f593pd0-corrscope-0.8.1-test-version.drv'...
env XDG_DATA_HOME=$TMPDIR /nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/bin/corr --version returned a non-zero exit code.
Matplotlib created a temporary config/cache directory at /private/tmp/nix-build-corrscope-0.8.1-test-version.drv-0/matplotlib-yq1cw7w_ because the default path (/homeless-shelter/.matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.
Matplotlib created a temporary config/cache directory at /private/tmp/nix-build-corrscope-0.8.1-test-version.drv-0/matplotlib-nbs9pqbj because the default path (/homeless-shelter/.matplotlib) is not a writable directory; it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.
Traceback (most recent call last):
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/Library/Application Support/corrscope'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/Library/Application Support'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/Library'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/bin/.corr-wrapped", line 6, in <module>
    from corrscope.cli import main
  File "/nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/lib/python3.10/site-packages/corrscope/cli.py", line 11, in <module>
    from corrscope.corrscope import template_config, CorrScope, Config, Arguments
  File "/nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/lib/python3.10/site-packages/corrscope/corrscope.py", line 12, in <module>
    from corrscope import outputs as outputs_
  File "/nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/lib/python3.10/site-packages/corrscope/outputs.py", line 10, in <module>
    from corrscope.settings.paths import MissingFFmpegError
  File "/nix/store/a30c1l4slgr2dmcji665jhdk42gqzblx-corrscope-0.8.1/lib/python3.10/site-packages/corrscope/settings/paths.py", line 23, in <module>
    appdata_dir.mkdir(parents=True, exist_ok=True)
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1179, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1179, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1179, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/nix/store/a4ivjgjcdk0l4yzylhc1q7acm7abdz6x-python3-3.10.10/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
OSError: [Errno 30] Read-only file system: '/homeless-shelter'
error: builder for '/nix/store/2vzb1xjr9hk6sjxf32j1ph856f593pd0-corrscope-0.8.1-test-version.drv' failed with exit code 1;
```

sigh